### ISSUE TITLE
mocha -w not reflecting element changes

### ISSUE THREAD

**Issue by cameronjroe** [airbnb/enzyme#169]
How would one use describeWithDOM() with `mocha -w`. When I make changes to my react component, the tests don't reflect on watch reload.

**Comment by ljharb** [comment]
@cameronjroe when you use `--watch`, the globals don't get wiped across test runs - thus the global `document` wouldn't either. Is there a chance you're making changes to `document` in your tests/implementation, without cleaning up `after`, that's persisting across runs?

**Comment by cameronjroe** [comment]
No, I'm just using `mount` within an `it` block. When I change the component the tests don't pick up the change.

```
it("should support additional props passed in", () => {
    const rating = mount(<StarRating name='rating' className='something' />);
    expect(rating.find('span').first().prop('className')).to.contain('rsr-container');
});
```

**Comment by cameronjroe** [comment]
Any ideas on this?

**Comment by ljharb** [comment]
I'm not certain, but one of the other contributors will be sure to get back to you, hopefully by Monday!

**Comment by cameronjroe** [comment]
Okay, cheers.

**Comment by lelandrichardson** [comment]
@cameronjroe that is strange. Can you describe more about your environment and your tests?  This seems like it might be a mocha issue. Is document provided by a browser or by jsdom?

**Comment by cameronjroe** [comment]
I'm using `describeWithDOM` which seems to be using jsdom. I'm requiring a jsx file of the component which doesn't reload once I start the tests. I'm also trying to run:

```
  after(() => {
    wrapper.unmount();
    wrapper.detach();
  })
```

but both methods are showing undefined.

**Comment by lelandrichardson** [comment]
@cameronjroe you might be using an older version of enzyme as those are newer methods. You also shouldn't need to run `.unmount()` or `.detach()`. The former is only useful if you want to test your component's `componentWillUnmount` lifecycle. The latter is only possible to be used if you are passing in an `attach` option to `mount()` which I don't believe you are.

**Comment by cameronjroe** [comment]
Okay, thanks. I just changed `.jsx` to `.js` and now the tests seem to run okay.
